python/pyspark/sql/dataframe.py

```diff
@@ -3067,6 +3067,10 @@ def transform(self, func: Callable[["DataFrame"], "DataFrame"]) -> "DataFrame":
         ----------
         func : function
             a function that takes and returns a :class:`DataFrame`.
+        args : Any
+            optional positional arguments for the func function
+        kwargs: Any
+            optional keyword arguments for the func function
```

Comment:
I think they are not the Parameters of `transform` ??

I think it should be described as a child of `func`.

e.g.
```python
        func : function
            a function that takes and returns a :class:`DataFrame`.
            * args : Any
                optional positional arguments for the func function
            * kwargs: Any
                optional keyword arguments for the func function
```

Reply:
> I think they are not the Parameters of `transform` ??

What I meant is the following (totally made up change);

- Let's say we want to add `foo` argument to `transform` i..e

    ```python
    def transform(self, func: Callable[["DataFrame"], "DataFrame"], foo: Any) -> "DataFrame"
    ```

- That's possible ATM, but if we add `*args` and `**kwargs` to the signature this becomes problematic as both
 
    ```python
   def transform(self, func: Callable[["DataFrame"], "DataFrame"], foo: Any, *args, **kwargs) -> "DataFrame"
    ```

   and

    ```python
   def transform(self, func: Callable[["DataFrame"], "DataFrame"], *args,  foo: Any,  **kwargs) -> "DataFrame"
    ```

   are limited and / or ambiguous.

Reply:
But like mentioned above ‒ that's not very likely scenario :)

Reply:
Ah, I got it.

Yeah, then it looks fine as it for now to me.